src/app/WriteClient.cpp

```diff
@@ -431,7 +431,11 @@ CHIP_ERROR WriteClient::OnMessageReceived(Messaging::ExchangeContext * apExchang
 
     if (mState == State::AwaitingTimedStatus)
     {
-        err = HandleTimedStatus(aPayloadHeader, std::move(aPayload));
+        CHIP_ERROR statusError = CHIP_NO_ERROR;
+        SuccessOrExit(err = StatusResponse::ProcessStatusResponse(std::move(aPayload), statusError));
```

Comment:
This needs to first check that the message is in fact a StatusResponse.

Reply:
yes, just add this check back. thanks for detailed check.